File: Java/RuleIntentDetector.java
package mu.cse.rag.clean;

import java.util.Locale;

public class RuleIntentDetector implements IntentDetector {
    @Override
    public Intent detect(String question) {
        String q = question.toLowerCase(Locale.forLanguageTag("tr-TR"));
        
        for (Intent intent : Intent.values()) {
            if (intent == Intent.UNKNOWN) continue;
                for (String keyword : intent.getKeywords()) {
                    if (q.contains(keyword)) {
                    return intent;
                    }
            }
        }
        return Intent.UNKNOWN;
    }
}

File: Python/RuleIntentDetector.py
from .Intent import Intent


class RuleIntentDetector:
    def detect(self, question: str) -> Intent:
        q = (question or '').lower()
        for intent in Intent:
            if intent == Intent.UNKNOWN:
                continue
            for keyword in intent.get_keywords():
                if keyword and keyword in q:
                    return intent
        return Intent.UNKNOWN

